loom/common/platform/platformNetwork.c

```diff
@@ -61,13 +66,68 @@ typedef int   SOCKET;
 
 lmDefineLogGroup(netLogGroup, "net", 1, LoomLogWarn);
 
+static const int readSpinSleepMs = 5;
+static const int readSpinTimeoutMs = 6000;
+static const int writeChunkMaxSize = 16384;
+static const int writeSmallCountThreshold = 200;
+static const int writeResizePagePower = 12;
+static const int writeShrinkThreshold = 3;
+static const int writeShrinkTarget = 2;
+
+// Structure keeping socket buffers and metadata
+struct SocketData {
+    // Handle to the socket itself
+    loom_socketId_t id;
+
+    // Write bip-buffer (modified circular buffer)
+    bipbuf_t* writeBuffer;
+
+    // The number of small writes made in a row.
+    // Used for identifying buffer shrink points.
+    int smallWrites;
+
+    // The number of failed writes made in a row.
+    // Used to detect stalled writing / unresponsive remote endpoint.
+    int stallWrites;
+
+    // 1 if write stall is detected, 0 otherwise
+    int stalled;
+
+    // Activity timer used in stall detection
+    loom_precision_timer_t activityTimer;
+
+    // The number of bytes sent, used for debug log reporting
+    unsigned int bytesSent;
+
+    // Required structure for the hash to work
+    UT_hash_handle hh;
+};
+
+// Head of the socket id -> SocketData hash table
+static struct SocketData* socketDatas = NULL;
```

Comment:
Yeah I didn't know what to call it and it got stuck. Socket table is good, maybe socket hash since it's really sort of a funky hash table.